infrastructure/storageprovider/dropbox.go

```diff
@@ -122,3 +148,29 @@ func (d *dropbox) prepareRequest(accessToken string, file io.Reader, fileName, s
 	req.Header.Set("Dropbox-API-Arg", dropboxAPIArg)
 	return req, nil
 }
+
+func (d *dropbox) mapToDropboxError(byteResponse *[]byte, httpStatusCode int) dropboxError {
+	strRes := string(*byteResponse)
+	errorRes := dropboxError{HttpCode: httpStatusCode}
+	errorRes.Message = strRes
+	if httpStatusCode == http.StatusUnauthorized {
+		errorRes.Json = dropboxErrorJson{}
+		json.Unmarshal(*byteResponse, &errorRes.Json)
+	}
+
+	return errorRes
+}
+
+func (d *dropbox) mapToRegularError(dropboxError *dropboxError) error {
```

Comment:
```suggestion
func (d *dropbox) mapToRegularError(dropboxError dropboxError) error {
```

no need to use pointer as it can't be nil anyway

Reply:
Okay